performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/MainView.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  Copyright 2021 Google LLC
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SwiftUI
+
+struct ImageView: View {
+  @ObservedObject var process: Process
+
+  var body: some View {
+    VStack {
+      if let uiImage = process.image {
+        Image(uiImage: uiImage).padding(.bottom)
+        if process.action == .classify {
+          if process.categories.isEmpty {
+            Button("Classify Image") {
+              if #available(iOS 15, tvOS 15, *) {
+                #if swift(>=5.5)
+                  Task { await process.classifyImageAsync() }
+                #endif
```

Comment:
My understanding is that this would make it necessary to use Xcode 13 even if compiling for iOS/tvOS 14.x, which is supported. Our CI wouldn't be able to compile the project either. I wanted to avoid the duplication with the else branch of the `#available` iOS/tvOS version check, but I can simply let the else branch of the `#if` swift version check be `process.classifyImage()` which I imagine would be less disruptive, so I'll take this approach unless you suggest otherwise.

Reply:
Nah, let's just leave it as-is. Thanks for clarifying!

Reply:
We would indeed have a problem with the current code if for some reason we are compiling for iOS/tvOS 15 but using Xcode 12 (which I don't think is possible?) or setting Xcode 13 to compile using lower versions of Swift (I thought it too unlikely), so I'll just add the extra else branch just in case.